Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/SharedLogic/Allocator/Jobs/Overlay/SendOverlay.hs

```diff
@@ -38,72 +42,95 @@ sendOverlayToDriver ::
 
-deleteNPaymentPendingDriverIds :: (CacheFlow m r, EsqDBFlow m r) => Id DM.Merchant -> Integer -> m ()
-deleteNPaymentPendingDriverIds merchantId num = Hedis.lTrim (makePaymentPendingDriverIdsKey merchantId) num (-1)
+deleteNSendOverlaySchedulerDriverIds :: (CacheFlow m r, EsqDBFlow m r) => Id DM.Merchant -> Id AnyJob -> Integer -> m ()
+deleteNSendOverlaySchedulerDriverIds merchantId jobId num = Hedis.lTrim (makeSendOverlaySchedulerDriverIdsKey merchantId jobId) num (-1)
 
-addPaymentPendingDriverIds :: (CacheFlow m r, EsqDBFlow m r) => Id DM.Merchant -> NonEmpty (Id DP.Person) -> m ()
-addPaymentPendingDriverIds merchantId = Hedis.rPush (makePaymentPendingDriverIdsKey merchantId)
+addSendOverlaySchedulerDriverIds :: (CacheFlow m r, EsqDBFlow m r) => Id DM.Merchant -> Id AnyJob -> NonEmpty (Id DP.Person) -> m ()
+addSendOverlaySchedulerDriverIds merchantId jobId = Hedis.rPush (makeSendOverlaySchedulerDriverIdsKey merchantId jobId)
 
-makePaymentPendingDriverIdsKey :: Id DM.Merchant -> Text
-makePaymentPendingDriverIdsKey merchantId = "SendOverlayScheduler:PaymentPendingDriverIds:merchantId-" <> merchantId.getId
+makeSendOverlaySchedulerDriverIdsKey :: Id DM.Merchant -> Id AnyJob -> Text
+makeSendOverlaySchedulerDriverIdsKey merchantId jobId = "SendOverlayScheduler:merchantId-" <> merchantId.getId <> ":jobId-" <> jobId.getId
 
-getBatchedDriverIds :: (CacheFlow m r, EsqDBFlow m r) => Id DM.Merchant -> m [Id DP.Person]
-getBatchedDriverIds merchantId = do
-  driverIdsLength <- getPaymentPendingDriverIdsLength merchantId
+getBatchedDriverIds :: (CacheFlow m r, EsqDBFlow m r) => Id DM.Merchant -> Id AnyJob -> DOverlay.OverlayCondition -> m [Id DP.Person]
+getBatchedDriverIds merchantId jobId condition = do
+  driverIdsLength <- getSendOverlaySchedulerDriverIdsLength merchantId jobId
+  transporterConfig <- SCT.findByMerchantId merchantId >>= fromMaybeM (TransporterConfigNotFound merchantId.getId)
   when (driverIdsLength < 1) do
-    drivers <- QDI.fetchAllDriversWithPaymentPending merchantId
-    whenJust (nonEmpty (drivers <&> (.driverId))) $ addPaymentPendingDriverIds merchantId
-  batchedDriverIds <- getFirstNPaymentPendingDriverIds merchantId 50
-  deleteNPaymentPendingDriverIds merchantId 50
+    driverIds <- case condition of
+      DOverlay.PaymentOverdueGreaterThan _ -> QDI.fetchAllDriversWithPaymentPending merchantId <&> (<&> (.driverId))
+      DOverlay.PaymentOverdueBetween _ _ -> QDI.fetchAllDriversWithPaymentPending merchantId <&> (<&> (.driverId))
+      DOverlay.InvoiceGenerated paymentMode -> do
+        now <- getLocalCurrentTime transporterConfig.timeDiffFromUtc
+        let potentialStartToday = addUTCTime (transporterConfig.driverPaymentCycleStartTime) (UTCTime (utctDay now) (secondsToDiffTime 0))
+            startTime = addUTCTime (-1 * transporterConfig.driverPaymentCycleDuration) potentialStartToday
+            endTime = addUTCTime 120 potentialStartToday
```

Comment:
Why 120?

Reply:
2 mins buffer because we want to fetch all driver fees that were created between range startTime and endTime but, there might me a case endTime is not matching because of few milli seconds. That's why.

Reply:
Won't endTime be the same always? How will it differ? Fine anyway